src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderHistory/CurrentOrdersController.ts

```diff
@@ -57,6 +66,16 @@ module Orckestra.Composer {
                                 .fail(() => {
                                     this.Loading = false;
                                 });
+                        },
+                        reload(){
+                            window.location.reload()
+                        },
+                        cancelOrderConfirm(event: JQueryEventObject, orderNumber: string) {
+                            this.OrderNumber = orderNumber;
+                            this.Modal.cancelOrderModal.openModal(event);
+                        },
+                        cancelOrder(){
+                            self.orderService.cancelOrder(this.OrderNumber); 
```

Comment:
add before 
 if(this.Loading) return;
      this.Loading = true;
on fail add 
.fail(() => {
                                    this.Loading = false;
                                });
                        },